docs/src/pages/ThemingSpecification.mdx

```diff
@@ -0,0 +1,297 @@
+export const meta = {
+  title: 'Theming specification',
+}
+
+## Goals and requirements
+
```

Comment:
I don't have an answer for this one, but it is a critical issue:

#### App developers can get small bundles when only using a subset of components.

I think the core difference between how Fabric and Stardust approached styling was this:

Fabric exports Checkbox and it comes with styles. If you use the Checkbox, you only get the Checkbox styles. You can define tokens in theme (colors, fonts, etc) which can get mapped into the styles. This means that bundle only contains applicable code, theme is small, and dev experience is nice (just use the Button.) You can even use Customizer to merge new styles on top of the defaults.

With the Stardust approach, styles come with the theme only. All components are "base" by default. Benefit being that you can completely hot swap the styling itself, the tech used to inject, tokens, anything, from the theme, without changing your app code. Unfortunately, decoupling this means you lose the treeshaking by default, where just using "uberTheme" with rendering the Button means you get the Button code, plus the styles for everything in uberTheme.

To make things smaller, you end up having to manually three shake your theme. It's possible to maybe do some webpack magic to shake out unused styling, but it's major work for us and puts a pretty big burden on the end user.

I haven't come up with any other solutions than basically this:

We can provide a BaseButton. It doesn't care about stying.

For M365, we can provide a Button which does the Fabric thing for better tree shaking and better dev exp. Theme only has tokens, no styling.

Reply:
Thanks for bringing this up! Will add this in the goals section, this is really something that we need to address. The manual tree shaking was what I had in mind, but I see the point of knowing what the component depends on. Although this may seem like a very complicated matter, you still have to know this if you were styling this component, as you probably have some styles applied on the slots too... I am not sure that adding styles together with the component is the right things to do, as if we do that, the styles are no longer part of the theme, and we will break the concept of composing themes (like base Fluent -> Teams theme overrides) - I understand that we can have some mechanisms for adding style overrides, but I believe we will easily hit the wall where we will have styles across different component's version, and will be hard to keep track what's where if they are not organized in a centralized place. Anyway, I will add this in the goals section and we can iterate on it.